helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -497,6 +500,77 @@ public boolean isInMaintenanceMode(String clusterName) {
         .exists(keyBuilder.maintenance().getPath(), AccessOption.PERSISTENT);
   }
 
+  @Override
+  public void setClusterManagementMode(ClusterManagementModeRequest request) {
+    ClusterManagementMode.Type mode = request.getMode();
+    String clusterName = request.getClusterName();
+    String reason = request.getReason();
+
+    // TODO: support other modes
+    switch (mode) {
+      case CLUSTER_PAUSE:
+        enableClusterPauseMode(clusterName, request.isCancelPendingST(), reason);
+        break;
+      case NORMAL:
+        // If from other modes, should check what mode it is in and call the api accordingly.
+        // If we put all mode config in one znode, one generic method is good enough.
+        disableClusterPauseMode(clusterName);
+        break;
+      default:
+        throw new IllegalArgumentException("ClusterManagementMode " + mode + " not supported");
+    }
+  }
+
+  private void enableClusterPauseMode(String clusterName, boolean cancelPendingST, String reason) {
+    String hostname = NetworkUtil.getLocalhostName();
+    logger.info(
+        "Enable cluster pause mode for cluster: {}. CancelPendingST: {}. Reason: {}. From Host: {}",
+        clusterName, cancelPendingST, reason, hostname);
+
+    BaseDataAccessor<ZNRecord> baseDataAccessor = new ZkBaseDataAccessor<>(_zkClient);
+    HelixDataAccessor accessor = new ZKHelixDataAccessor(clusterName, baseDataAccessor);
+
+    if (baseDataAccessor.exists(accessor.keyBuilder().pause().getPath(), AccessOption.PERSISTENT)) {
+      throw new HelixConflictException(clusterName + " pause signal already exits");
+    }
+    if (baseDataAccessor.exists(accessor.keyBuilder().maintenance().getPath(), AccessOption.PERSISTENT)) {
+      throw new HelixConflictException(clusterName + " maintenance signal already exits");
+    }
+
+    PauseSignal pauseSignal = new PauseSignal();
+    pauseSignal.setPauseCluster(Boolean.toString(true));
```

Comment:
nit, Can we make it in the constructor parameter to avoid possible misusage?

Ultimately, it would be great if we have a super class for all signals. And the 3 modes are child classes.
Then on set, directly create child class objects.
On read, the super class can have a static method to convert into the child class according to the internal mode field.
This would help to avoid possible misusage and duplicate code.